dspace-uclouvain/src/main/java/org/dspace/uclouvain/pdfAttestationGenerator/handlers/MasterThesisPdfAttestationGeneratorHandler.java

```diff
@@ -0,0 +1,136 @@
+package org.dspace.uclouvain.pdfAttestationGenerator.handlers;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+import javax.xml.transform.Result;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.sax.SAXResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.apache.fop.apps.FOUserAgent;
+import org.apache.fop.apps.Fop;
+import org.apache.fop.apps.FopFactory;
+import org.apache.fop.apps.MimeConstants;
+import org.dspace.content.Bitstream;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.uclouvain.core.utils.MetadataUtils;
+import org.dspace.uclouvain.pdfAttestationGenerator.configuration.PDFAttestationGeneratorConfiguration;
+import org.dspace.uclouvain.pdfAttestationGenerator.exceptions.PDFGenerationException;
+import org.dspace.uclouvain.pdfAttestationGenerator.model.MasterThesisPDFAttestationModel;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/** 
+* Main handler to generate an attestation for a thesis
+*/
+public class MasterThesisPdfAttestationGeneratorHandler implements PDFAttestationGeneratorHandler {
+
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    MetadataUtils metadataUtils;
+
+    @Autowired
+    PDFAttestationGeneratorConfiguration config;
+
+    private String source = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("dspace.dir");
+
+    /** 
+    * Recovers data about the object and uses it to construct the PDF file.
+    * 
+    * @param response: The response object from the controller. The PDF file will be passed to it.
+    * @param uuid: UUID of the targeted DSpace object.
+    */
+    public void getAttestation(OutputStream out, UUID uuid) throws PDFGenerationException {
+        try {
+            Context DSpaceContext = new Context();
+
+            File templateFile = new File(this.source + "/config/pdf/attestation/" + config.getConfigForItemType("Thesis").templateName);
```

Comment:
Is this file could be define in a custom dspace configuration key ?
Something like `DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("dspace.pdfAttestation.templateDir");`